### PR TITLE
#5368 Remove non-breaking space character on footer

### LINKED ISSUE #5368: Don't add "--" separator if user signature begins with "--"
FlowCrypt keeps adding "--" at the beginning of my signature. Which is unfortunate, because my signature already begins with "-- ". It's pretty ugly.

I believe correct behavior would be to not add anything to the signature?

### PR DESCRIPTION
**Pull request by martgil**
This PR removes the non-breaking space character in the email footer/signature that causes a duplicate footer separator.

close #5368 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Test added or updated.

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/chrome/elements/compose-modules/compose-footer-module.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class ComposeFooterModule extends ViewModule<ComposeView> {
   public createFooterHtml = (footer: string) => {
     const sanitizedPlainFooter = Xss.htmlSanitizeAndStripAllTags(footer, '\n');
     const sanitizedHtmlFooter = sanitizedPlainFooter.replace(/\n/g, '<br>');
-    const footerFirstLine = sanitizedPlainFooter.split('\n')[0];
+    const footerFirstLine = sanitizedPlainFooter.split('\n')[0].replace(/&nbsp;/g, '');
```
Comment (sosnovsky) [inline extension/chrome/elements/compose-modules/compose-footer-module.ts:37]: let's replace it with common space, so it'll be ` ` instead of empty string

**Review by sosnovsky — COMMENTED** [review]
Well done, it fixes issue on my side👍

And let's add a test which will check that there is no double separator in this case. I think it should be a simple one - you can set user's signature by using `configProvider.google.aliases` config

File: extension/chrome/elements/compose-modules/compose-footer-module.ts (hunk 1 above)
Comment (martgil) [reply extension/chrome/elements/compose-modules/compose-footer-module.ts:37]: Thank you - will do.

> And let's add a test which will check that there is no double separator in this case. I think it should be a simple one - you can set user's signature by using configProvider.google.aliases config

I agree - I'll be writing the test now. thank you.

Comment (martgil) [reply extension/chrome/elements/compose-modules/compose-footer-module.ts]: Roma, for me the `configProvider.google.aliases` doesn't fit very well on the test so I try to re-use the idea of directly putting a signature with `configProvider.google.acctPrimarySignature` which is great for the test. I hope that is also an acceptable approach.

**Comment by martgil** [comment]
ready for review. thank you!

**Review by sosnovsky — APPROVED** [review]
Well done 👍

**Pull request merged**